src/Features/Core/Portable/DesignerAttribute/DesignerAttributeDiscoveryService.cs

```diff
@@ -121,9 +120,20 @@ private static async ValueTask<bool> HasDesignerCategoryTypeAsync(Project projec
                 }
 
                 // Handle the priority doc first.
-                var priorityDocument = solution.GetDocument(priorityDocumentId);
-                if (priorityDocument != null)
-                    await ProcessProjectAsync(priorityDocument.Project, priorityDocument, useFrozenSnapshots, callback, cancellationToken).ConfigureAwait(false);
+                if (priorityDocumentId != null)
+                {
+                    // Create a frozen snapshot guaranteed to have this document in it.
+                    var priorityDocument = solution
+                        .GetRequiredDocument(priorityDocumentId)
+                        .WithFrozenPartialSemantics(cancellationToken);
+
+                    await ProcessProjectAsync(priorityDocument.Project, priorityDocument, callback, cancellationToken).ConfigureAwait(false);
+                }
+
+                // Freeze the entire solution at this point.  We don't want to run generators (as they are very unlikely
+                // to contribute any changes that would affect which types we think are designable), and we want to be 
+                // very fast to update the ui as a user types.
+                solution = await solution.WithFrozenPartialCompilationsAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
If we're going to freeze evrything anyways, are we best off doing this up front and use that for the specific project case? Or is the overhead here big enough that we're worried this will slow down the "priority" too much?

Reply:
freezing everything doesn't guarantee we have the priority document in the frozen solution.  so i very much want to stil say "freeze, but ensure this document is in there" and "freeze everything".  Note, this took us down to 2-3 calls instead of hundreds.  So i think the main win is already met.

Reply:
Good point about the document not guaranteed to be in afterwards -- might be worth a comment.

Reply:
i'll add comment.